src/ci_ex_states/excited_state_solver.cc

```diff
@@ -0,0 +1,1131 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ * @END LICENSE
+ */
+
+#include <fstream>
+#include <iomanip>
+#include <cmath>
+
+#include "psi4/libpsi4util/PsiOutStream.h"
+#include "psi4/physconst.h"
+#include "excited_state_solver.h"
+#include "sci/sci.h"
+#include "base_classes/mo_space_info.h"
+#include "sparse_ci/operator.h"
+#include "helpers/helpers.h"
+#include "helpers/printing.h"
+#include "ci_rdm/ci_rdms.h"
+namespace forte {
+ExcitedStateSolver::ExcitedStateSolver(StateInfo state, size_t nroot,
+                                       std::shared_ptr<MOSpaceInfo> mo_space_info,
+                                       std::shared_ptr<ActiveSpaceIntegrals> as_ints,
+                                       std::unique_ptr<SelectedCIMethod> sci)
+    : ActiveSpaceMethod(state, nroot, mo_space_info, as_ints), sci_(std::move(sci)) {
+    nact_ = mo_space_info_->size("ACTIVE");
+}
+
+void ExcitedStateSolver::set_options(std::shared_ptr<ForteOptions> options) {
+    // TODO: This shouldn't come from options
+    root_ = options->get_int("ROOT");
+
+    ex_alg_ = options->get_str("ACI_EXCITED_ALGORITHM");
+    core_ex_ = options->get_bool("ACI_CORE_EX");
+    if (options->has_changed("ACI_QUIET_MODE")) {
+        quiet_mode_ = options->get_bool("ACI_QUIET_MODE");
+    }
+    add_singles_ = options->get_bool("ACI_ADD_SINGLES");
+    direct_rdms_ = options->get_bool("ACI_DIRECT_RDMS");
+    test_rdms_ = options->get_bool("ACI_TEST_RDMS");
+    save_final_wfn_ = options->get_bool("ACI_SAVE_FINAL_WFN");
+    first_iter_roots_ = options->get_bool("ACI_FIRST_ITER_ROOTS");
+    sparse_solver_ = std::make_shared<SparseCISolver>(as_ints_);
+    sparse_solver_->set_parallel(true);
+    sparse_solver_->set_force_diag(options->get_bool("FORCE_DIAG_METHOD"));
+    sparse_solver_->set_e_convergence(options->get_double("E_CONVERGENCE"));
+    sparse_solver_->set_maxiter_davidson(options->get_int("DL_MAXITER"));
+    sparse_solver_->set_spin_project(options->get_bool("ACI_PROJECT_OUT_SPIN_CONTAMINANTS"));
+    sparse_solver_->set_spin_project_full(options->get_bool("ACI_PROJECT_OUT_SPIN_CONTAMINANTS"));
+    sparse_solver_->set_guess_dimension(options->get_int("DL_GUESS_SIZE"));
+    sparse_solver_->set_num_vecs(options->get_int("N_GUESS_VEC"));
+    sparse_solver_->set_sigma_method(options->get_str("SIGMA_BUILD_TYPE"));
+    sparse_solver_->set_max_memory(options->get_int("SIGMA_VECTOR_MAX_MEMORY"));
```

Comment:
We should think about changing the option names with ACI to SCI or something more general

Reply:
Yes, but currently in order not to change the input files of aci and pci test cases dramatically, we can do this cleanup in new PRs.

Reply:
That makes sense